Unable to use webpack

### Environment

Nuxt project info:                                                                                                                                                                                                                    00:21:10

------------------------------
- Operating System: `Linux`
- Node Version:     `v14.19.3`
- Nuxt Version:     `3.0.0`
- Nitro Version:    `1.0.0`
- Package Manager:  `npm@6.14.17`
- Builder:          `webpack`
- User Config:      `builder`
- Runtime Modules:  `-`
- Build Modules:    `-`
------------------------------

### Reproduction

Setup

```bash
npx nuxi init nuxt-app
cd nuxt-app
npm install && npm install webpack @nuxt/webpack-builder
```

Enable webpack:

```js
// https://nuxt.com/docs/api/configuration/nuxt-config
export default defineNuxtConfig({
    builder: 'webpack',
})
```

### Describe the bug

I'm unable to use webpack in a default install.

### Additional context

_No response_

### Logs

```shell
> nuxt build

Nuxi 3.0.0                                                                                                                                                                                                                            00:20:07
Nuxt 3.0.0 with Nitro 1.0.0                                                                                                                                                                                                           00:20:07

 ERROR  Nuxt build error                                                                                                                                                                                                              00:20:10

  Module not found: Error: Can't resolve '#build/paths.mjs' in 'node_modules/nuxt/dist/app'
  
  ERROR in ./node_modules/nuxt/dist/app/entry.mjs 5:0-20
  Module not found: Error: Can't resolve '#build/css' in 'node_modules/nuxt/dist/app'
  
  ERROR in ./node_modules/nuxt/dist/app/entry.mjs 6:0-38
  Module not found: Error: Can't resolve '#build/plugins' in 'node_modules/nuxt/dist/app'
  
  ERROR in ./node_modules/nuxt/dist/app/entry.mjs 7:0-54
  Module not found: Error: Can't resolve '#build/root-component.mjs' in 'node_modules/nuxt/dist/app'
  
  ERROR in ./node_modules/nuxt/dist/app/entry.mjs 8:0-51
  Module not found: Error: Can't resolve '#build/nuxt.config.mjs' in 'node_modules/nuxt/dist/app'
  
  client compiled with 5 errors
```

--- a/packages/webpack/src/presets/base.ts
+++ b/packages/webpack/src/presets/base.ts
@@ -158,7 +158,8 @@ export function baseTranspile (ctx: WebpackConfigContext) {
   const transpile = [
     /\.vue\.js/i, // include SFCs in node_modules
     /consola\/src/,
-    /vue-demi/
+    /vue-demi/,
+    /(^|\/)nuxt\/(dist\/)?app($|\/)/
   ]
 
   for (let pattern of options.build.transpile) {

